pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1559,6 +1559,12 @@ void receiveIndividualMessagesFromBatch(BrokerEntryMetadata brokerEntryMetadata,
                 }
                 if (possibleToDeadLetter != null) {
                     possibleToDeadLetter.add(message);
+                    // when redeliveryCount > deadLetterPolicy.getMaxRedeliverCount())
+                    // the message will not be visible to users
+                    if (redeliveryCount > deadLetterPolicy.getMaxRedeliverCount()) {
+                        continue;
```

Comment:
skippedMessages++;

Reply:
We should make the change at https://github.com/apache/pulsar/pull/17326/files#diff-f6e4c1c4091aa10525f331e48e66b29f22b9f7987755c1b4fb887e24f198bed6R1539-R1541 ?

And looks like https://github.com/apache/pulsar/pull/17326/files#diff-f6e4c1c4091aa10525f331e48e66b29f22b9f7987755c1b4fb887e24f198bed6R1579-R1588 should also move to  https://github.com/apache/pulsar/pull/17326/files#diff-f6e4c1c4091aa10525f331e48e66b29f22b9f7987755c1b4fb887e24f198bed6R1539-R1541